src/test/regress/expected/multi_view.out

```diff
@@ -268,3 +268,514 @@ SELECT * FROM  lineitems_by_orderkey WHERE l_orderkey = 100;
+      90 | Sun Jan 12 21:37:30.778206 2014 |      11 |     458 |     377 |        
+     100 | Sun Jan 19 22:32:08.284043 2014 |       2 |     384 |     149 |        
+(18 rows)
+
+-- determine if a recent user has done a given event type or not
+SELECT ru.user_id, CASE WHEN et.user_id IS NULL THEN 'NO' ELSE 'YES' END as done_event
+	FROM recent_users ru LEFT JOIN events_table et ON(ru.user_id = et.user_id AND et.event_type = 625) ORDER BY 2 DESC, 1;
+ user_id | done_event 
+---------+------------
+       6 | YES
+      13 | NO
+      39 | NO
+      44 | NO
+      46 | NO
+      50 | NO
+      58 | NO
+      66 | NO
+      71 | NO
+      74 | NO
+      86 | NO
+      87 | NO
+      90 | NO
+     100 | NO
+(14 rows)
+
+-- create a select only view
+CREATE VIEW selected_users AS SELECT * FROM users_table WHERE value_1 >= 120 and value_1 <150;
+CREATE VIEW recent_selected_users AS SELECT su.* FROM selected_users su JOIN recent_users ru USING(user_id);
+SELECT user_id FROM recent_selected_users GROUP BY 1 ORDER BY 1;
+ user_id 
+---------
+       6
+      13
+      39
+      44
+      46
+      50
+      58
+      66
+      71
+      74
+      86
+      90
+(12 rows)
+
+-- this would be supported when we implement where partition_key in (subquery) support
+SELECT et.* FROM events_table et WHERE et.user_id IN (SELECT user_id FROM recent_selected_users);
+ERROR:  could not run distributed query with subquery outside the FROM clause
+HINT:  Consider using an equality filter on the distributed table's partition column.
+-- it is supported if it is a router query
+SELECT count(*) FROM events_table et WHERE et.user_id IN (SELECT user_id FROM recent_selected_users WHERE user_id = 90);
+ count 
+-------
+   109
+(1 row)
+
+-- expected this to work but it did not
+(SELECT user_id FROM recent_users) 
+UNION
+(SELECT user_id FROM selected_users);
+ERROR:  could not run distributed query with UNION, INTERSECT, or EXCEPT
+HINT:  Consider using an equality filter on the distributed table's partition column.
+-- wrapping it inside a SELECT * works
+SELECT *
+	FROM (
+		(SELECT user_id FROM recent_users) 
+		UNION
+		(SELECT user_id FROM selected_users) ) u
+	WHERE user_id < 15 AND user_id > 10
+	ORDER BY user_id;
+ user_id 
+---------
+      11
+      12
+      13
+      14
+(4 rows)
+
+-- union all also works for views
+SELECT *
+	FROM (
+		(SELECT user_id FROM recent_users) 
+		UNION ALL
+		(SELECT user_id FROM selected_users) ) u
+	WHERE user_id < 15 AND user_id > 10
+	ORDER BY user_id;
+ user_id 
+---------
+      11
+      11
+      11
+      12
+      12
+      12
+      12
+      12
+      12
+      13
+      13
+      13
+      13
+      13
+      14
+(15 rows)
+
+SELECT count(*)
+	FROM (
+		(SELECT user_id FROM recent_users) 
+		UNION
+		(SELECT user_id FROM selected_users) ) u
+	WHERE user_id < 15 AND user_id > 10;
+ count 
+-------
+     4
+(1 row)
+
+-- expected this to work but it does not
```

Comment:
Yes, you're right this may seem to be expected to work considering that the same query works with the relations. However, this is not a bug, I'd call this as a limitation. 

I think we should improve top level union/union all support in the next release and ensure that this test pass.